notes/connection-lifecycle.md

```diff
@@ -6,78 +6,125 @@
 **first origin being connected to** to reach the target origin. This either means
 the target origin itself or the proxy origin if one is desired.
 
-If the connection is an `HTTPSConnection`, then `HTTPSConnection.set_cert()` should
-be called once with information about how to verify certificates. (What is stopping us
-from putting this information in `HTTPSConnection.__init__()`?)
+```python
+import urllib3.connection
+
+# Initialize the HTTPSConnection ('https://...')
+conn = urllib3.connection.HTTPSConnection(
+    host="example.com",
+    # Here you can configure other options like
+    # 'ssl_minimum_version', 'ca_certs', etc.
+)
+
+# Set the connect timeout either in the
+# constructor above or via the property.
+conn.timeout = 3.0  # (connect timeout)
+```
 
 If using CONNECT tunneling with the proxy, call `HTTPConnection.set_tunnel()`
-with the tunneled host, port, and headers. This will mark the `HTTPConnection`
-as not-reusable.
+with the tunneled host, port, and headers. This should be called before calling
+`HTTPConnection.connect()` or sending a request.
+
+```python
+conn = urllib3.connection.HTTPConnection(
+    # Remember that the *first* origin we want to connect to should
+    # be configured as 'host' and 'port', *not* the target origin.
+    host="myproxy.net",
+    port=8080,
+    proxy="http://myproxy.net:8080"
+)
 
-Set the connection's `timeout` property to the "connect timeout" value.
+conn.set_tunnel("example.com", scheme="http", headers={"Proxy-Header": "value"})
+```
 
 Connect to the first origin by calling the `HTTPConnection.connect()` method.
 If an error occurs here you can check whether the error occurred during the
-connection to the proxy if `HTTPConnection._connecting_to_proxy` is true.
+connection to the proxy if `HTTPConnection.has_connected_to_proxy` is true.
 If the value is false then the error didn't occur while connecting to a proxy.
```

Comment:
So if `HTTPConnection.has_connected_to_proxy` is `False`, you haven't connected to the proxy, and the error occurred while connecting to it? (It seems that `_connecting_to_proxy` and `has_connected_to_proxy` are opposites so the wording should change.)